src/beautifiers/rubocop.coffee

```diff
@@ -16,53 +18,50 @@ module.exports = class Rubocop extends Beautifier
       rubocop_path: true
   }
 
-  beautify: (text, language, options) ->
+  beautify: (text, language, options, context) ->
+    fullPath = context.filePath
+    [projectPath, relativePath] = atom.project.relativizePath(fullPath)
+
+    # Find the rubocop path
+    rubocopPath = "rubocop"
     @Promise.all([
       @which(options.rubocop_path) if options.rubocop_path
       @which('rubocop')
-    ]).then((paths) =>
+    ])
+    .then((paths) =>
       @debug('rubocop paths', paths)
-      _ = require 'lodash'
-      path = require 'path'
       # Get first valid, absolute path
-      rubocopPath = _.find(paths, (p) -> p and path.isAbsolute(p) )
+      rubocopPath = paths.find((p) -> p and path.isAbsolute(p))
       @verbose('rubocopPath', rubocopPath)
       @debug('rubocopPath', rubocopPath, paths)
+    )
 
-      configFile = path.join(atom.project.getPaths()[0], ".rubocop.yml")
-
-      fs = require 'fs'
+    # Find or generate a config file if non exists
+    if !@findFile(path.dirname(fullPath), ".rubocop.yml")?
+      yaml = require("yaml-front-matter")
+      config = {
+        "Style/IndentationWidth":
+          "Width": options.indent_size
+      }
+      tempConfig = @tempFile("rubocop-config", yaml.safeDump(config))
 
-      if fs.existsSync(configFile)
-        @debug("rubocop", config, fs.readFileSync(configFile, 'utf8'))
-      else
-        yaml = require("yaml-front-matter")
-        # Generate config file
-        config = {
-          "Style/IndentationWidth":
-            "Width": options.indent_size
-        }
+    rubocopArguments = [
+      "--auto-correct"
+      "--force-exclusion"
+      "--stdin", relativePath || temp.path({suffix: '.rb'})
```

Comment:
```coffeescript
relativePath || temp.path({suffix: '.rb'})
```

[The `relativePath` is the directory for the project containing the file.](https://atom.io/docs/api/v1.21.0/Project#instance-relativizePath)

In the case of the file being unsaved, and therefore the `relativePath` is `undefined` could this not simply be `undefined`? Note that `undefined` argument values will be removed automatically: https://github.com/Glavin001/atom-beautify/blob/master/src/beautifiers/executable.coffee#L230-L232

Is `temp.path({suffix: '.rb'})` necessary?

Reply:
Rubocop requires you to specify a file path when using `--stdin <file>`, so we have to put something there. It uses the filename for the formatted output (whether you want it or not). So when it's `undefined`, we'd get an error:

```
» cat test.rb | rubocop --stdin -f json -a
-s/--stdin requires exactly one path.
```

It looks like it doesn't actually matter what we put in there, so I can change it to something hard coded like "atom-beautify.rb", which will only appear in the output that we throw away and won't affect the result.

Reply:
I'm very impressed with how attentive you are to this code! No wonder this project is so well maintained.

Reply:
More *interesting* decisions made by Rubocop maintainers 😝. 

I did a little digging, and looks like it may be a new bug: https://github.com/bbatsov/rubocop/issues/4666#issuecomment-322635604

> It looks like it doesn't actually matter what we put in there, so I can change it to something hard coded like "atom-beautify.rb", which will only appear in the output that we throw away and won't affect the result.

Good idea. Let's go with it 👍 .

> I'm very impressed with how attentive you are to this code! No wonder this project is so well maintained.

Thank you! My plan for [Unibeautify](https://github.com/Unibeautify/unibeautify) is to be even better!